explainability-core/src/main/java/org/kie/trustyai/explainability/model/Dataframe.java

```diff
@@ -0,0 +1,688 @@
+     * @return Resulting {@link List} {@link Value}.
+     */
+    public List<Value> reduceRows(Function<List<Value>, Value> fn) {
+        return rowIndexStream().mapToObj(row -> fn.apply(getRow(row))).collect(Collectors.toList());
+    }
+
+    public void addColumn(String name, Type type, List<Value> values) {
+        data.add(new ArrayList<>(values));
+
+        metadata.names.add(name);
+        metadata.types.add(type);
+        metadata.constrained.add(true);
+        metadata.domains.add(EmptyFeatureDomain.create());
+        metadata.inputs.add(false);
+    }
+
+    public List<Prediction> asPredictions() {
+        final List<Integer> inputIndices = getInputsIndices();
+        final List<Integer> outputIndices = getOutputsIndices();
+        List<List<Feature>> allInputs = getInputsIndices().stream().map(this::columnAsFeatures).collect(Collectors.toList());
+        List<List<Output>> allOutputs = getOutputsIndices().stream().map(this::columnAsOutputs).collect(Collectors.toList());
+
+        List<Prediction> predictions = new ArrayList<>();
+        for (int row = 0; row < this.getRowDimension(); row++) {
+            List<Feature> features = new ArrayList<>();
+            for (int col = 0; col < inputIndices.size(); col++) {
+                features.add(allInputs.get(col).get(row));
+            }
+            List<Output> outputs = new ArrayList<>();
+            for (int col = 0; col < outputIndices.size(); col++) {
+                outputs.add(allOutputs.get(col).get(row));
+            }
+            PredictionInput input = new PredictionInput(features);
+            PredictionOutput output = new PredictionOutput(outputs);
+            predictions.add(new SimplePrediction(input, output));
+        }
+        return predictions;
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("Dataframe [").append(getRowDimension()).append("x").append(getColumnDimension()).append("]\n");
+        for (int i = 0; i < getColumnDimension(); i++) {
+            builder.append("\tColumn ").append(i).append(" (").append(metadata.names.get(i)).append(")\n");
+            builder.append("\t\tType: ").append(metadata.types.get(i)).append("\n");
+            builder.append("\t\tDomain: ");
+            final FeatureDomain domain = metadata.domains.get(i);
+            if (domain.isEmpty()) {
+                builder.append("(no domain)");
+            } else {
+                builder.append(domain.prettyPrint());
+            }
+            builder.append("\n");
+        }
+        return builder.toString();
+    }
+
+    private class Metadata {
+        private List<String> names = new ArrayList<>();
+        private List<Type> types = new ArrayList<>();
+        private List<Boolean> constrained = new ArrayList<>();
+        private List<FeatureDomain> domains = new ArrayList<>();
+        private List<Boolean> inputs = new ArrayList<>();
```

Comment:
Similar suggestion here
 
```suggestion
        private final List<String> names;
        private final List<Type> types;
        private final List<Boolean> constrained;
        private final List<FeatureDomain> domains;
        private final List<Boolean> inputs;
        
        private Metadata() {
            this(new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>(), new ArrayList<>());
        }
        
        private Metadata(List<String> names, List<Type> types, List<Boolean> constrained, List<FeatureDomain> domains, List<Boolean> inputs) {
            this.names = names;
            this.types = types;
            this.constrained = constrained;
            this.domains = domains;
            this.inputs = inputs;
        }
```